functions/utils.R

```diff
@@ -1,8 +1,62 @@
 
+# convert [list] parent_list: child_list1, child_list2 
+# to [vector]: child_list1 with name as child_list2 
 list2Vector <- function(list) {
-  vector <- c()
-  for (i in seq_along(list)) {
-    vector[list[[i]][[2]]] <- list[[i]][[1]]
+  
+  child <- sapply(list, `[[`, 1)
+  names(child) <- sapply(list, `[[`, 2)
+
+  return(child)
+}
+
+# get manifest info from getDatasetManifest()
+collectManifestInfo <- function(manifest_info) {
+
+  manifest <- NULL
+
+  if (manifest_info != "") {
+    manifest_path <- manifest_info["path"]
+    manifest_id <- manifest_info["id"]
+    create_date <- manifest_info["createdOn"]
+    modified_date <- manifest_info["modifiedOn"]
+
+    manifest_df <- data.table::fread(manifest_path)
+
+    if ("Component" %in% colnames(manifest_df) & nrow(manifest_df) > 0) {
+      
+      manifest_component <- manifest_df[["Component"]][1]
+      manifest <- list(list(manifest_id, manifest_component, create_date, modified_date, manifest_path))
+    }
   }
-  return(vector)
+  
+  return(manifest)
 }
+
+# get information of manifest from output list of collectManifestInfo()
+extractManifests <- function(list) {
+
+  list <- list[lengths(list) != 0]
+  df <- data.frame()
+
+  if (length(list) != 0) {
+    df <- data.frame(
+      synID = sapply(list, `[[`, c(1, 1)),
+      schema = sapply(list, `[[`, c(1, 2)),
+      create = sapply(list, `[[`, c(1, 3)) %>% as.Date(),
+      modify = sapply(list, `[[`, c(1, 4)) %>% as.Date(),
+      path = sapply(list, `[[`, c(1, 5))
+    ) %>% 
+      filter(schema != "" & schema != "NaN") %>% 
+      # distinct(schema, .keep_all = TRUE) %>% 
```

Comment:
add comment to indicate whether to use unique component or multiple manifest with the same component name.